config/utils.ts

```diff
@@ -0,0 +1,447 @@
+  let source = '';
+
+  try {
+    source = fs.readFileSync(configPath).toString();
+  } catch (err) {
+    throw new FileSystemError(
+      { cause: err },
+      {
+        filepath: configPath,
+        operation: 'read',
+      }
+    );
+  }
+
+  return source;
+}
+
+export function removeUndefinedFieldsFromConfigAccount<
+  T extends
+    | HubSpotConfigAccount
+    | Partial<HubSpotConfigAccount> = HubSpotConfigAccount,
+>(account: T): T {
+  Object.keys(account).forEach(k => {
+    const key = k as keyof T;
+    if (account[key] === undefined) {
+      delete account[key];
+    }
+  });
+
+  if ('auth' in account && account.auth) {
+    if (account.authType === OAUTH_AUTH_METHOD.value) {
+      Object.keys(account.auth).forEach(k => {
+        const key = k as keyof OAuthConfigAccount['auth'];
+        if (account.auth?.[key] === undefined) {
+          delete account.auth?.[key];
+        }
+      });
+    }
+
+    if (
+      'tokenInfo' in account.auth &&
+      typeof account.auth.tokenInfo === 'object'
+    ) {
+      Object.keys(account.auth.tokenInfo).forEach(k => {
+        const key = k as keyof TokenInfo;
+        if (account.auth?.tokenInfo[key] === undefined) {
+          delete account.auth?.tokenInfo[key];
+        }
+      });
+    }
+  }
+
+  return account;
+}
+
+// Ensure written config files have fields in a consistent order
+export function formatConfigForWrite(config: HubSpotConfig) {
+  const {
+    defaultAccount,
+    defaultCmsPublishMode,
+    httpTimeout,
+    allowUsageTracking,
+    accounts,
+    ...rest
+  } = config;
+
+  const orderedConfig = {
+    ...(defaultAccount && { defaultAccount }),
+    defaultCmsPublishMode,
+    httpTimeout,
+    allowUsageTracking,
+    ...rest,
+    accounts: accounts.map(account => {
+      const { name, accountId, env, authType, ...rest } = account;
+
+      return {
+        name,
+        accountId,
+        env,
+        authType,
+        ...rest,
+      };
+    }),
+  };
+
+  return removeUndefinedFieldsFromConfigAccount(orderedConfig);
+}
+
+export function writeConfigFile(
+  config: HubSpotConfig,
+  configPath: string
+): void {
+  const source = yaml.dump(
+    JSON.parse(JSON.stringify(formatConfigForWrite(config), null, 2))
+  );
+
+  try {
+    fs.ensureFileSync(configPath);
+    fs.writeFileSync(configPath, source);
+  } catch (err) {
+    throw new FileSystemError(
+      { cause: err },
+      {
+        filepath: configPath,
+        operation: 'write',
+      }
+    );
+  }
+}
+
+function getAccountType(sandboxAccountType?: string): AccountType {
+  if (sandboxAccountType) {
+    if (sandboxAccountType.toUpperCase() === 'DEVELOPER') {
+      return HUBSPOT_ACCOUNT_TYPES.DEVELOPMENT_SANDBOX;
+    }
+    if (sandboxAccountType.toUpperCase() === 'STANDARD') {
+      return HUBSPOT_ACCOUNT_TYPES.STANDARD_SANDBOX;
+    }
+  }
+  return HUBSPOT_ACCOUNT_TYPES.STANDARD;
+}
+
+export function normalizeParsedConfig(
```

Comment:
This function reads both old and new config formats and converts them to an object with the same shape that can be used around LDL and the CLI

(for example, converts `portals` to `accounts` among other things)